schema/module_schema.go

```diff
@@ -113,6 +115,17 @@ func schemaForDependentModuleBlock(localName string, modMeta *module.Meta) (*sch
 		}
 	}
 
+	if strings.HasPrefix(module.SourceAddr, "registry.terraform.io/") {
+		shortName := strings.TrimPrefix(module.SourceAddr, "registry.terraform.io/")
```

Comment:
As mentioned in https://github.com/hashicorp/terraform-ls/pull/872#pullrequestreview-941954314 I didn't realize while discussing this earlier that it won't be as simple as using `strings.HasPrefix` here - mostly because I didn't understand why it wasn't working in the first place until you pointed out the 1.1.0 change.

We need to account for older Terraform versions too.

I'd be fine with the same solution I proposed in the linked comment, assuming we can create an issue for `tfregistry.ParseRawModuleSourceString()` which is now needed in two places.

Reply:
https://github.com/hashicorp/terraform-registry-address/pull/7 enables parsing the module source address via `tfaddr.ParseRawModuleSourceRegistry`. I've updated this PR accordingly